Source/Orts.Simulation/Simulation/RollingStocks/SubSystems/CruiseControl.cs

```diff
@@ -0,0 +1,1458 @@
+                                    controllerVolts += step;
+                                }
+                            }
+                            if (a > 0 && (SpeedIsMph ? MpS.ToMpH(Locomotive.WheelSpeedMpS) : MpS.ToKpH(Locomotive.WheelSpeedMpS)) > 5)
+                            {
+                                if (controllerVolts > demandedVolts && Locomotive.AccelerationMpSS > a + 0.02)
+                                {
+                                    float step = 100 / ThrottleFullRangeIncreaseTimeSeconds;
+                                    step *= elapsedClockSeconds;
+                                    controllerVolts -= step;
+                                }
+                            }
+                            else
+                            {
+                                if (controllerVolts > demandedVolts)
+                                {
+                                    float step = 100 / ThrottleFullRangeIncreaseTimeSeconds;
+                                    step *= elapsedClockSeconds;
+                                    controllerVolts -= step;
+                                }
+                            }
+                            if (controllerVolts > demandedVolts && delta < 0.8)
+                            {
+                                float step = 100 / ThrottleFullRangeIncreaseTimeSeconds;
+                                step *= elapsedClockSeconds;
+                                controllerVolts -= step;
+                            }
+                        }
+                        if (a > 0 && (SpeedIsMph ? MpS.ToMpH(Locomotive.WheelSpeedMpS) : MpS.ToKpH(Locomotive.WheelSpeedMpS)) > 5)
+                        {
+                            if ((a != Locomotive.AccelerationMpSS) && delta > 0.8)
+                            {
+                                if (Locomotive.AccelerationMpSS < a + 0.02)
+                                {
+                                    float step = 100 / ThrottleFullRangeIncreaseTimeSeconds;
+                                    step *= elapsedClockSeconds;
+                                    controllerVolts += step;
+                                }
+                                if (Locomotive.AccelerationMpSS > a - 0.02)
+                                {
+                                    float step = 100 / ThrottleFullRangeIncreaseTimeSeconds;
+                                    step *= elapsedClockSeconds;
+                                    controllerVolts -= step;
+                                }
+                            }
+                        }
+
+                        if (UseThrottle)
+                        {
+                            //Simulator.Confirmer.Information(Locomotive.Train.AccelerationMpSpS.SmoothedValue.ToString());
+                            // 
+                            if (SpeedRegMode == SpeedRegulatorMode.AVV)
```

Comment:
I would prefer if this block was moved to a TCS script, keeping in this class a more generic Automatic Train Operation (ATO) system.

I suggest the TCS-ATO interface to be like this:
```csharp
void SetATOTargetParameters(float targetSpeedMpS, float targetDistanceM)
```
I think the targetDistanceM is necessary to accurately stop a train in a platform when required, but I might be wrong.

Reply:
Removed